src/openforms/contrib/objects_api/clients/objects.py

```diff
@@ -15,3 +15,17 @@ def create_object(self, objecttype_url: str, record_data: dict) -> dict:
         response.raise_for_status()
 
         return response.json()
+
+    def update_object(
+        self, objecttype_url: str, record_data: dict, initial_data_reference: str
```

Comment:
I think in this case we can use `object_uuid` instead of `itinal_data_reference` (as it is scoped to the Objects API)